File: HoWo5_6/ShopService/Shipment.java
package HoWo5_6.ShopService;

import java.util.ArrayList;

import HoWo5_6.Phone.Phone;

public class Shipment implements IntrfTransaction {

    @Override
    public void changingTheBalance(Phone phone, PhoneMag mag) {
        ArrayList<Phone> priceList = new ArrayList<>();
        priceList = mag.getPriceList();
        if (mag.getMoney() >= phone.getPrice()){
            mag.setMoney(mag.getMoney()-phone.getPrice());

            phone.setPrice(phone.getPrice()*1.1);
            priceList.add(phone);
            mag.setPriceList(priceList);
        }
        else{
            PrintMessage.printMassage("Недостаточно средств!");
        }
    }


}

File: HoWo7/ShopService/Shipment.py
from Phone import Phone
from ShopService import PhoneMag
from ShopService import PrintMassage


class Shipment:
    @staticmethod
    def changingTheBalance( phone:Phone,  mag:PhoneMag):
        priceList = mag.getPriceList()
        if mag.getMoney() >= phone.getPrice():
            mag.setMoney(mag.getMoney()-phone.getPrice())

            phone.setPrice(phone.getPrice()*1.1)
            priceList.append(phone)
            mag.setPriceList(priceList)
        else:
            PrintMassage.printMassage("Недостаточно средств!")